server/portal/apps/intromessages/views.py

```diff
@@ -0,0 +1,42 @@
+"""
+.. :module: apps.intromessages.views
+   :synopsis: Views to handle read/unread status of IntroMessages
+"""
+
+import logging
+from portal.views.base import BaseApiView
+from django.http import JsonResponse
+from portal.apps.intromessages.models import IntroMessages
+from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+import json
+
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(login_required, name='dispatch')
+class IntroMessagesView(BaseApiView):
+    # Get all of the IntroMessages that have been read (unread = False)
+    def get(self, request, *args, **kwargs):
+        messages_array = IntroMessages.objects.filter(user=request.user).values('component', 'unread').values()
+        messages = [{'component': message['component'], 'unread': message['unread']} for message in messages_array]
+        return JsonResponse({ 'response': messages})
+
+
+    def put(self, request, *args):
+        body = json.loads(request.body)     # IntroMessages data from front end 
+        # get all of the IntroMessages stored in the database (meaning that those messages have been read/dismissed) for the user
+        db_messages = IntroMessages.objects.filter(user=request.user).values()
+        for component_name, component_value in body.items():
+            if component_value['unread'] == False:      # if message has been read/dismissed
+                found = False
```

Comment:
We did discuss this a little, but I thought we decided to leave it just writing when unread is False.  The idea being less writes to the database and less changes to the front end saga.  Since IntroMessages is likely changing, we talked about more work on the saga in a separate Jira ticket.  Nonetheless, I'm happy to do whatever you or Sal (other reviewer) think is best.

Reply:
It wouldn't change the frontend saga, right?  Or am I misunderstanding something.

Reply:
Well, it would really change the redux store as the IntroMessages' status are updated in the store as one group.

Reply:
Maybe I'm missing something but if the endpoint created or updated entries with both true and false for `unread` then our responses could have a true and false like for example
```
[{"component": "COMPONENT1", "unread": False}, {"component": "COMPONENT2", "unread": True}]
```
then saga (as is) would still produce a redux state that is the same as it is currently now and still compatible with the frontend:
```
{"COMPONENT1": False, "COMPONENT2": True}
```

but I still may not be following things. maybe talk on slack?

Reply:
OK.  That's totally doable.  The question would then be, when would you do the initial write to the database?  Would you write all of the IntroMessages to the database with unread = True the first time the user logs in after this change goes into production?

Reply:
Right now (with no additional) changes because you setup the state to have each component as `unread=true`, they would all be written when the user updated just one of them (by clicking dismiss).  So need to worry about things like prepping the db beforehand or during login.

Reply:
As recommended, intromessages are now written to the database with `unread = true` or `unread = false`.